lib/fog/openstack/core.rb

```diff
@@ -572,6 +572,33 @@ def self.get_supported_version(supported_versions, uri, auth_token, connection_o
       version
     end
 
+    def self.get_supported_version_path(supported_versions, uri, auth_token, connection_options = {})
```

Comment:
this is awesome, could you put this code also for keystone v3?

I've hit an issue that keystone endpoint-list doesn't have to contain url for right version, so it's safer to discover it.